src/commands/cmd_query.c

```diff
@@ -75,6 +75,12 @@ void _MGraph_Query(void *args) {
     }
     CommandCtx_ThreadSafeContextUnlock(qctx);
 
+    bool reply_compact = false;
+    // TODO race condition?
```

Comment:
When a client is unblocked, its `argv` array is freed, which is why `CommandCtx_New` duplicates the graph name.

As such, I think we might want to make this comparison in the critical region just above, but it might be unnecessary (hence the note).